File: answers/leetcode/Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
		int price = 0;
		for (int i = 1; i < prices.size(); ++ i)
		{
			if (prices[i] > prices[i - 1])
			{
				price += prices[i] - prices[i - 1];
			}
		}
		return price;
    }
};

// greedy
// 2014-09-06
class Solution {
public:
	int maxProfit(vector<int> &prices) {
		size_t buy = 0;
		int sum = 0;
		for (size_t i = 1; i < prices.size(); ++ i)
		{
			if (prices[i] > prices[buy])
			{
				sum += prices[i] - prices[buy];
			}
			buy = i;
		}
		return sum;
    }
};

File: answers/leetcode/Best Time to Buy and Sell Stock II/Best Time to Buy and Sell Stock II.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices == null || prices.length == 0) {
            return 0;
        }
        int profit = 0;
        int buy = prices[0];
        for (int price : prices) {
            if (price > buy) {
                profit += price - buy;
            }
            buy = price;
        }
        return profit;
    }
}